src/structure.js

```diff
@@ -68,6 +69,34 @@ Structure.prototype.cursor = function (path) {
   return Cursor.from(self.current, path, changeListener);
 };
 
+Structure.prototype.reference = function (path, fn) {
+  var self = this;
+  fn = fn || function () { };
+
+  var cursor = this.cursor(path);
+  var changeListener = function (newRoot, oldRoot, changedPath) {
+    fn(newRoot, oldRoot, path);
+    cursor = self.cursor(path);
+  };
+  this.on('swap', changeListener);
+
+  return {
+    onChange: function (newFn) {
```

Comment:
Will need a way to "unsubscribe" functions. Otherwise, repeated calls to `componentDidMount` will attach functions, which will be repeated executed, causing an unnecessary amounts of cascading of function calls.

I'm basing this on this example:

```js
var clock = quiz.reference('clock');

let Clock = component('Clock', [ forceUpdateOn(clock) ], () =>
  div({},
    String(clock.cursor().deref())));

function forceUpdateOn (ref) {
  return {
    componentDidMount: function () {
      ref.onChange(_ => this.forceUpdate());
    },
    componentWillUnmount: function () {
      ref.unsubscribe();
    }
  };
}
```